gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java

```diff
@@ -462,10 +474,47 @@ private void createOrUpdateMembers(final ExecutionContext executionContext, ApiE
                 if (roleIdsToImport.contains(poRoleId)) {
                     futurePo = memberToImport;
                 }
+
+                /*
+                 * When the API comes from Kubernetes, we remove the members which are not in use anymore
+                 * If the same behaviour required for all APIs regardless of their origin, then the context check
+                 * can be removed
+                 */
+                if (apiEntity.getOriginContext().isOriginKubernetes()) {
+                    // by removing the member, we will eventually end up with a clean list of
+                    // Members that can be removed from this API
+                    membersAlreadyPresent.remove(memberToImport);
+                }
+            }
+
+            if (apiEntity.getOriginContext().isOriginKubernetes()) {
```

Comment:
Not sure but I think we should check the origin on the apiJSONNode. If we enter an update from a CRD export, I'm not sure at all the apiEntity will have a kubernetes origin